ner_v2/detectors/temporal/date/date_detection.py

```diff
@@ -0,0 +1,706 @@
+            detector_constant.DATE_TO_PROPERTY: to_property,
+            detector_constant.DATE_START_RANGE_PROPERTY: start_range_property,
+            detector_constant.DATE_END_RANGE_PROPERTY: end_range_property,
+            detector_constant.DATE_NORMAL_PROPERTY: normal_property,
+            detector_constant.DATE_DETECTION_METHOD: detection_method,
+        }
+
+    def _date_model_detection(self):
+        """
+        This function calls run_model functionality from class Models() and verifies the values returned by it through
+        datastore.
+        If the dates provided by the model are present in the datastore, it sets the value to FROM_MODEL_VERIFIED
+        else FROM_MODEL_NOT_VERFIED is set.
+
+        For Example:
+            Note:  before calling this method you need to call set_bot_message() to set a bot message.
+
+            self.bot_message = 'Please help me with your departure date?'
+            self.text = '26th november'
+
+            Output:
+               [
+                 {
+
+                  'detection_method': 'message',
+                  'end_range': False,
+                  'from': False,
+                  'normal': True,
+                  'start_range': False,
+                  'text': '26th november',
+                  'to': False,
+                  'value': {'dd': 26, 'mm': 11, 'type': 'date', 'yy': 2017}
+                  'detection_method': model_verified
+
+                  }
+               ]
+
+
+
+        For Example:
+
+            self.bot_message = 'Please help me with your departure date?'
+            self.text = 'monday'
+
+            Output:
+                 [
+                    {
+
+                      'detection_method': 'message',
+                      'end_range': False,
+                      'from': False,
+                      'normal': True,
+                      'start_range': False,
+                      'text': 'monday',
+                      'to': False,
+                      'value': {'dd': 21, 'mm': 8, 'type': 'day_within_one_week', 'yy': 2017}
+                      'detection_method': model_not_verified
+
+                    }
+                 ]
+
+        """
+        date_dict_list = []
+        model_object = Models()
+        model_output = model_object.run_model(entity_type=model_constant.DATE_ENTITY_TYPE,
+                                              bot_message=self.bot_message, user_message=self.text)
+        for output in model_output:
+            entity_value_list, original_text_list = self._date_value(text=output[model_constant.MODEL_DATE_VALUE])
+            if entity_value_list:
+                date_value = entity_value_list[0]
+                detection_method = FROM_MODEL_VERIFIED
+            else:
+                date_value = output[model_constant.MODEL_DATE_VALUE]
+                detection_method = FROM_MODEL_NOT_VERIFIED
+
+            date_dict_list.append(
+                {
```

Comment:
Use this function '_to_output_dict' instead of making dictionary over here. Easy to maintain if output dict gets generated from one place.

Reply:
done